lib/spells/BattleSpellMechanics.cpp

```diff
@@ -436,15 +436,17 @@ bool ObstacleMechanics::isHexAviable(const CBattleInfoCallback * cb, const Battl
 	if(cb->battleGetStackByPos(hex, true))
 		return false;
 
-	auto obst = cb->battleGetObstacleOnPos(hex, false);
-	if(obst /*&& obst->type != CObstacleInstance::MOAT*/)//todo: issue 2366, uncomment once obstacle tile sharing implemented
-		return false;
+	auto obst = cb->battleGetAllObstaclesOnPos(hex, false);
+
+	for(auto & i : obst)
+		if(i->obstacleType != CObstacleInstance::MOAT)
+			return false;
 
-	if(nullptr != cb->battleGetDefendedTown() && CGTownInstance::NONE != cb->battleGetDefendedTown()->fortLevel())
+	if(cb->battleGetDefendedTown() != nullptr && cb->battleGetDefendedTown()->fortLevel() != CGTownInstance::NONE)
```

Comment:
Since Alex okay with it I wouldn't say against this change, but I want to hear your opinion on that.
From my point of view that's not a bad way to prevent game breaking typos.

Probably we should decide whatever comparison like that should be used or not.